src/GitHub.VisualStudio/GitHubPackage.cs

```diff
@@ -45,24 +46,23 @@ public GitHubPackage(IServiceProvider serviceProvider)
             this.serviceProvider = serviceProvider;
         }
 
-        protected override void Initialize()
+        protected override async tasks.Task InitializeAsync(CancellationToken cancellationToken, IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
-            IncrementLaunchCount();
+            var usageTracker = await GetServiceAsync(typeof(IUsageTracker)) as IUsageTracker;
+            await tasks.Task.Run(() => usageTracker.IncrementLaunchCount());
```

Comment:
The deadlock is probably unrelated. In any case, the `InitializeAsync` method is definitely getting called on the main thread in most cases here.